src/Button/Button.js

```diff
@@ -71,34 +80,96 @@ const StyledButton = createComponent({
+      padding: 0 16px;
       font-size: ${fontSize}px;
       width: ${block ? '100%' : 'auto'};
-      background: ${outline || transparent ? 'transparent' : backgroundColor};
-      border: ${transparent ? 'none' : `1px solid ${backgroundColor}`};
+      background: ${outline ? 'transparent' : backgroundColor};
+      border-color: ${outline ? backgroundColor : borderColor || backgroundColor};
+      border-style: solid;
+      border-width: 1px;
       transition: 175ms;
+      white-space: nowrap;
+      outline: none;
+      user-select: none;
 
-      ${loading && loadingCss({ height, fontColor, outline, backgroundColor })};
+      ${icon &&
+        hasText &&
+        css`
+          padding-left: 8px;
+
+          i {
+            padding-right: 4px;
+          }
+        `}
+
+      ${loading && loadingCss({ height, fontColor, backgroundColor, borderColor, outline })};
 
       &:hover {
-        background: ${outline ? 'transparent' : lighten(0.05, backgroundColor)};
-        border-color: ${lighten(0.05, backgroundColor)};
+        background-color: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${hover &&
+          css`
+            background-color: ${hover.backgroundColor || backgroundColor};
+            border-color: ${hover.borderColor || hover.backgroundColor || backgroundColor};
+            color: ${hover.fontColor || fontColor};
+          `}
       }
 
       &:active {
-        background: ${outline ? 'transparent' : lighten(0.075, backgroundColor)};
-        border-color: ${lighten(0.075, backgroundColor)};
+        background: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${active &&
+          css`
+            background-color: ${active.backgroundColor || backgroundColor};
+            border-color: ${active.borderColor || active.backgroundColor || backgroundColor};
+            color: ${active.fontColor || fontColor};
+          `}
+      }
+
+      &[disabled] {
+        pointer-events: none;
+        background: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${disabledState &&
+          css`
+            background-color: ${disabledState.backgroundColor || backgroundColor};
+            border-color: ${disabledState.borderColor || disabledState.backgroundColor || backgroundColor};
+            color: ${disabledState.fontColor || fontColor};
+          `}
       }
 
+      ${text &&
+        css`
+          background-color: transparent !important;
+          border-color: transparent !important;
+          color: ${textColor || backgroundColor} !important;
+        `}
+
       ${space};
     `;
   },
 });
 
-const Button = React.forwardRef((props, ref) => <StyledButton {...props} ref={ref} />);
+const Button = React.forwardRef((props, ref) => {
+  const hasText = !!props.children;
```

Comment:
No need to set this extra variable - can look at `props.children` directly in `StyledButton`